wzdx/combination/attenuator.py

```diff
@@ -0,0 +1,190 @@
+import json
+from datetime import datetime
+from wzdx.tools import cdot_geospatial_api, polygon_tools
+import logging
+
+ATTENUATOR_TIME_AHEAD_SECONDS = 30 * 60
+ISO_8601_FORMAT_STRING = "%Y-%m-%dT%H:%M:%SZ"
+
+
+def get_route_info_planned_event(planned_event):
+    start = cdot_geospatial_api.get_route_and_measure(
+        planned_event['geometry']['coordinates'][0])
+    end = cdot_geospatial_api.get_route_and_measure(
+        planned_event['geometry']['coordinates'][1])
+    if start['Route'] != end['Route'] or start['Measure'] == end['Measure']:
+        return None
+    else:
+        return {
+            'route': start['Route'],
+            'start_measure': start['Measure'],
+            'end_measure': end['Measure']
+        }
+
+
+def get_route_info_geotab(geotab):
+    start = cdot_geospatial_api.get_route_and_measure(
+        geotab['geometry']['coordinates'][0])
+    end = cdot_geospatial_api.get_route_and_measure(
+        geotab['geometry']['coordinates'][1])
+    if start['Route'] != end['Route'] or start['Measure'] == end['Measure']:
+        return None
+    else:
+        return {
+            'route': start['Route'],
+            'start_measure': start['Measure'],
+            'end_measure': end['Measure']
+        }
+
+
+def json_serial(obj):
+    """JSON serializer for objects not serializable by default json code"""
+
+    if isinstance(obj, (datetime)):
+        return obj.strftime(ISO_8601_FORMAT_STRING)
+    raise TypeError("Type %s not serializable" % type(obj))
+
+
+def main():
+    with open('./wzdx/sample_files/raw/geotab_avl/geotab_all.json') as f:
```

Comment:
Are these just hard-coded for testing? I assume we plan on pulling from a live feed for the data to combine?

Reply:
these are just hardcoded, yeah. This script should never be called from the command line, only as methods though